db-mojo-scorer/README.md

```diff
@@ -0,0 +1,258 @@
+    5. Classpath: The license is loaded from resource called '/license.sig' The default resource name can be changed via system property 'ai.h2o.mojos.runtime.license.filename'
+    ```
+       
+- To run the database scorer, you can directly run the executable jar:
+    ```
+    java -Dpropertiesfilename={PATH_TO_PROPERTIES_FILE} -jar build/libs/dai-mojo-db-{YOUR_CURRENT_VERSION}.jar
+    ```
+
+### Properties Files
+
+Using the flag -Dpropertiesfilename= multiple configuration files can be created for different models or databases, just specify the properties file to use at runtime.
+
+#### Sample Properties File
+
+```
+ModelName=pipeline.mojo
+SQLConnectionString=jdbc:postgresql://192.168.1.171:5432/LendingClub SQLUser=postgres
+SQLPassword=aDJvaDJvCg==
+SQLPrompt=
+SQLKey=id
+SQLPrediction=
+SQLSelect=select id, loan_amnt, term, int_rate, installment, emp_length, home_ownership, annual_inc, verification_status, addr_state, dti, delinq_2yrs, inq_last_6mths, pub_rec, revol_bal, revol_util, total_acc from "import".loanstats4
+SQLWrite=update "import".loanstats4 set where id=
+```
+The above will score using **pipeline.mojo** and use the key 'id' to update the column name from the mojo into the table, to `bad_loan.0` and `bad_loan.1` must be in the table schema.
+
+If SQLPrediction was set to ModelPrediction (for example) then `bad_loan.0` would be written to the table column ModelPrediction.
+
+
+#### Usage
+```
+java -Dpropertiesfilename=DAI-Mojo-DB/properties/DAIMojoRunner_DB.properties-Snowflake -jar DAI-Mojo-DB/build/libs/dai-mojo-db-2.30.jar
+```
+
+### Runtime Flags
+
+While the Database and Mojo configuration is in the properties file, the command line does have some flags
+
+ - `-Dverbose=true` | false{default}       
+      verbosity flag
+ - `-Doverride=true`| false{default}     
+      bypass check if running multithreads on a table without an index, as the table would be in a random order.
+ - `-Dwait=true`| false: {default}       
+      causes the program to wait for Enter once the model and threads are ready, used for demos.
+ - `-Dstats=true` | false{default}  
+      writes performance statistics at completion.
+ - `-Dinspect=true`| false{default}  
+      prints model details and suggest SQLSelect based on model as well as runtime java memory settings.
+ - `-Dcapacity=` Number of available Cores+33%{default}  
+     Size of buffer queue.
+ - `-Dthreads=` Number of available Cores{default}
+ - `-Dpropertiesfilename=DAIMojoRunner_DB.properties`{default}        
+      allows multiple properties files to be created for different models and databases
+
+Example:
+```
+java -Derrors=true -Dverbose=true -Dthreads=2 -Dcapacity=4000 -Xms2g -Xmx10g -Dpropertiesfilename={PATH_TO_PROPERTIES_FILE} -jar build/libs/dai-mojo-db-{YOUR_CURRENT_VERSION}.jar
+```
+
+## Deployment Tricks
+
+The option `-Dinspect=true` helps, see this example. Assumes the model is called `pipeline.mojo` but could be change using the properties file.
```

Comment:
The option -Dinsepect=true can be used to generate the SQL based on the model, this helps when a large number of database columns are required.

Its important to also have a key for each row being scored, this is because the scorer will execute in parallel (based on the -Dthreads= parameter) so updating the databases will be faster if a unique key is used for each row (use a customer id or even the row number) for the SQLKey statement.